component-runtime-beam/src/main/java/org/talend/sdk/component/runtime/beam/spi/record/AvroRecord.java

```diff
@@ -112,6 +115,10 @@ private Object directMapping(final Object value) {
         if (value instanceof Date) {
             return Date.class.cast(value).getTime();
         }
+        if (value instanceof Instant) {
+            //TCOMP-2293 support Instant with nano second
+            return new long[]{Instant.class.cast(value).getEpochSecond(), Instant.class.cast(value).getNano()};
```

Comment:
avro record works for cloud platform for ser/desr for spark/beam, and tck DATETIME will use this avro schema:
https://github.com/Talend/component-runtime/blob/117e2c64296c7fec809598606e8cb4ef784373ca/component-runtime-beam/src/main/java/org/talend/sdk/component/runtime/beam/spi/record/AvroSchemaBuilder.java#L55
so guess this long[] not match that, and real can ser/deser for cloud platform?

Reply:
Why not extend with an union? Thus, we'll have union like {null, long, long[]}. WDYT @wwang-talend?

Reply:
Hi @undx , LGTM, no better idea now